app/code/Magento/Catalog/Api/CategoryLinkRepositoryInterface.php

```diff
@@ -46,4 +46,28 @@ public function delete(\Magento\Catalog\Api\Data\CategoryProductLinkInterface $p
      * @throws \Magento\Framework\Exception\StateException
      */
     public function deleteByIds($categoryId, $sku);
+
+    /**
+     * Replace category products
+     *
+     * @param int $categoryId
+     * @param \Magento\Catalog\Api\Data\CategoryProductLinkInterface[] $productLinks
+     * @return bool will returned True if assigned
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \Magento\Framework\Exception\StateException
+     */
+    public function replace($categoryId, $productLinks);
```

Comment:
We can't add/remove any methods in API interfaces according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/). 
In your case better to use our [bulk API framework](https://github.com/magento/bulk-api-ce), that designed especially for work with multiple operations.

Reply:
@sidolov 

I understand that it is maybe smart to use the bulk API but this is no backward compatible issue because it is for 2.3 or am I mistaking?

Reply:
@lewisvoncken unfortunately, we can't change interfaces marked as @api for 2.3Magento\Customer\CustomerData\SectionPool::getSectionsData, according to our rules